src/ctypes/ldouble.mli

```diff
@@ -0,0 +1,258 @@
+    [format fmt ldbl] will be used in C as [sprintf(..., fmt, ldbl)]. *)
+
+val to_string : t -> string
+(** Convert a long double to a string *)
+
+val of_string : string -> t
+(** Create a long double from a string *)
+
+(* debugging only *)
+(*val to_hex_string : t -> string*)
+
+val add : t -> t -> t 
+(** Addition *)
+
+val sub : t -> t -> t 
+(** Subtraction *)
+
+val mul : t -> t -> t 
+(** Multiplication *)
+
+val div : t -> t -> t 
+(** Division *)
+
+val neg : t -> t
+(** Negation *)
+
+val pow : t -> t -> t 
+(** Exponentiation *)
+
+val sqrt : t -> t 
+(** Square root *)
+
+val exp : t -> t 
+(** Exponential *)
+
+val log : t -> t 
+(** Natural logarithm *)
+
+val log10 : t -> t 
+(** Base 10 logarithm *)
+
+val expm1 : t -> t 
+(** [expm1 x] computes [exp x -. 1.0], giving numerically-accurate results
+    even if [x] is close to [0.0].
+*)
+
+val log1p : t -> t 
+(** [log1p x] computes [log(1.0 +. x)] (natural logarithm),
+    giving numerically-accurate results even if [x] is close to [0.0].
+    @since 3.12.0
+*)
+
+val cos : t -> t 
+(** Cosine.  Argument is in radians. *)
+
+val sin : t -> t 
+(** Sine.  Argument is in radians. *)
+
+val tan : t -> t 
+(** Tangent.  Argument is in radians. *)
+
+val acos : t -> t 
+(** Arc cosine *)
+
+val asin : t -> t 
+(** Arc sine *)
+
+val atan : t -> t 
+(** Arc tangent *)
+
+val atan2 : t -> t -> t 
+(** [atan2 y x] returns the arc tangent of [y /. x]. *)
+
+val hypot : t -> t -> t 
+
+val cosh : t -> t 
+(** Hyperbolic cosine *)
+
+val sinh : t -> t 
+(** Hyperbolic sine *)
+
+val tanh : t -> t 
+(** Hyperbolic tangent *)
+
+val acosh : t -> t 
+(** Inverse hyperbolic cosine *)
+
+val asinh : t -> t 
+(** Inverse hyperbolic sine *)
+
+val atanh : t -> t 
+(** Inverse hyperbolic tangent *)
+
+val ceil : t -> t 
+(** Round above to an integer value. *)
+
+val floor : t -> t 
+(** Round below to an integer value. *)
+
+val abs : t -> t 
+(** [abs f] returns absolute value of [f] *)
+
+val rem : t -> t -> t
+(** [rem x y] is the remainder of dividing x by y *)
+
+val copysign : t -> t -> t  
+(** [copysign x y] returns a float whose absolute value is that of [x]
+  and whose sign is that of [y]. *)
+
+val frexp : t -> t * int
+(** [frexp f] returns the pair of the significant and the exponent of [f]. *)
+
+val ldexp : t -> int -> t
+(** [ldexp x n] returns [x *. 2 ** n]. *)
+
+val modf : t -> t * t
+(** return [(fractional,integer)] parts of number.
+ 
+    Known fatal bug on mingw32; see https://sourceforge.net/p/mingw-w64/bugs/478 *)
+
+val classify : t -> fpclass
+(** Return the class of the given floating-point number:
+   normal, subnormal, zero, infinite, or not a number. *)
+
+val min_float : t
+(** The smallest positive, non-zero, non-denormalized value *)
+
+val max_float : t
+(** The largest positive finite value *)
+
+val epsilon : t
+(** The difference between [1.0] and the smallest exactly representable
+    floating-point number greater than [1.0]. *)
+
+val nan : t
```

Comment:
There appear to be multiple `nan` values for `ldouble`, which isn't the case for from OCaml's `float` type:

```ocaml
	 # Ldouble.(compare (neg nan) nan);;
	 - : int = 1
	 # (compare (-.nan) nan);;
	 - : int = 0
```